runtime/src/cost_tracker.rs

```diff
@@ -215,10 +221,12 @@ impl CostTracker {
         for account_key in keys.iter() {
             match self.cost_by_writable_accounts.get(account_key) {
                 Some(chained_cost) => {
-                    if chained_cost.saturating_add(cost) > self.account_cost_limit {
-                        return Err(CostTrackerError::WouldExceedAccountMaxLimit);
-                    } else {
-                        continue;
+                    if let Some(account_cost_limit) = self.account_cost_limit {
```

Comment:
I'd move this `if let Some(account_cost_limit) = self.account_cost_limit` out to be above the `for`. Then we skip the whole iteration is there's no limit.